disk/src/com/treode/disk/Checkpointer.scala

```diff
@@ -36,15 +36,18 @@ private class Checkpointer (kit: DiskKit) {
 
   private def reengage() {
     fanout (checkreqs) .pass (())
-    checkreqs = List.empty
-    bytes = 0
-    entries = 0
-    engaged = false
+    if (config.checkpoint (this.bytes, this.entries))
+      _checkpoint()
+    else
+      checkreqs = List.empty
+      engaged = false
```

Comment:
I got a chance to look closely at this CL this morning.  Given your exposure to Python, your assumption here is reasonable, but it's wrong. :-)

    if (...)
       then-clause
    else
       else-clause
       oops

Python (and Haskell, and others) would use the indentation to infer that In `oops` is part of the else clause. However Scala does not, and the above code is like this:

    if (...) {
        then-clause
    } else {
        else-clause
    }
    oops

Could you fix that. Set `engaged` to false in the else clause. Leave the assignment to `checkreqs` where it originally was.